src/ilcdlib/common.py

```diff
@@ -206,15 +208,31 @@ def remap_xml_ns(self, doc_ns_map: dict[str, str]) -> None:
             if url and url.endswith("EPD/2019"):  # Some providers use outdated, non-standard namespace
                 self.xml_parser.xml_ns["epd2019"] = url
 
+    def get_xml_for_entity(
+        self, provider: BaseIlcdMediumSpecificReader, entity_type: str, entity_id: str, entity_version: str | None
+    ) -> T_ET.Element | None:
+        """Attempt to fetch XML for given entity details and return it if successful."""
```

Comment:
Maybe make it part of ITB dialect, not general flow?

Reply:
Updated